healthcheck/health_check.go

```diff
@@ -0,0 +1,83 @@
+package HealthCheck
+
+import (
+	"context"
+	"time"
+)
+
+type Checker interface {
+	CheckAppHealth(ctx *context.Context) (check *Check, err error)
+}
+
+type Check struct {
+	Name        string    `json:"name"`
+	Status      string    `json:"status"`
+	StatusCode  int       `json:"status_code"`
+	Message     string    `json:"message"`
+	LastChecked time.Time `json:"last_checked"`
+	LastSuccess time.Time `json:"last_success"`
+	LastFailure time.Time `json:"last_failure"`
+}
+
+type HealthResponse struct {
+	Status    string        `json:"status"`
+	Version   string        `json:"version"`
+	Uptime    time.Duration `json:"uptime"`
+	StartTime time.Time     `json:"start_time"`
+	Checks    []*Check      `json:"checks"`
+}
+
+type HealthCheck struct {
+	Clients                   []*Client
```

Comment:
I am not a huge fan of this circular reference here, 
`HealthCheck` contains a list of `Clients`
`HealthCheck` contains a list of `Tickers`
`Ticker` contains a reference to a `Client` and kind of acts as a wrapper for the `Client`

... Tempted to remove `Clients` from the `HealthCheck` and just store them individually in the `ticker`s.

^ If this approach was taken the handler would get a little more bloated, as it would need to reach in like:
`ticker.client.check` instead of `client.check`.